src/Microsoft.Identity.Web/Policy/ScopeAuthorizationHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    ///  Scope authorization handler that needs to be called for a specific requirement type.
+    ///  In this case, <see cref="ScopeAuthorizationRequirement"/>.
+    /// </summary>
+    internal class ScopeAuthorizationHandler : AuthorizationHandler<ScopeAuthorizationRequirement>
+    {
+        private readonly IConfiguration _configuration;
+
+        /// <summary>
+        /// Constructor for the scope authorization handler, which takes a configuration.
+        /// </summary>
+        /// <param name="configuration">Configuration.</param>
+        public ScopeAuthorizationHandler(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        ///  Makes a decision if authorization is allowed based on a specific requirement.
+        /// </summary>
+        /// <param name="context">AuthorizationHandlerContext.</param>
+        /// <param name="requirement">Scope requirement.</param>
+        /// <returns>Task.</returns>
+        protected override Task HandleRequirementAsync(
+            AuthorizationHandlerContext context,
+            ScopeAuthorizationRequirement requirement)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            // The resource is either the HttpContext or the Endpoint directly when used with the
+            // authorization middleware
+            var endpoint = context.Resource switch
+            {
+                HttpContext httpContext => httpContext.GetEndpoint(),
+                Endpoint ep => ep,
+                _ => null,
+            };
+
+            var data = endpoint?.Metadata.GetMetadata<IAuthRequiredScopeMetadata>();
+
+            IEnumerable<string>? scopes = null;
+            if (requirement?.RequiredScopesConfigurationKey != null)
```

Comment:
What is the desired behavior if _both_ the config key and a set of allowed values are passed to the `ctor` for `ScopeAuthorizationRequirement`? It looks to me like the config will win completely and the ctor value will be ignored. Is that what's desired?